ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -470,8 +486,15 @@ void ActiveProvider_Disconnect(VirtualizationRootHandle rootIndex)
 
         VirtualizationRoot* root = &s_virtualizationRoots[rootIndex];
         assert(nullptr != root->providerUserClient);
+        assertf(userClient == root->providerUserClient, "ActiveProvider_Disconnect: disconnecting provider IOUC %p for root index %d (%s), but expecting IOUC %p",
+            KextLog_Unslide(userClient), rootIndex, root->path, KextLog_Unslide(root->providerUserClient));
         
         assert(NULLVP != root->rootVNode);
+        
+        KextLog_FileNote(root->rootVNode, "ActiveProvider_Disconnect: disconnecting provider (PID %d, IOUC %p) for virtualization root %d: (path: \"%s\", fsid: 0x%x:%x, inode: 0x%llx) directory vnode %p:%u.",
```

Comment:
Could you remind me, will `KEXTLOG_NOTE` level logging make it into the OS log?  Or is that only INFO level or higher?

Reply:
* `KEXTLOG_INFO`, corresponding to  `os_log`'s `OS_LOG_TYPE_INFO`, doesn't hit disk unless followed by an ERROR.
* `KEXTLOG_NOTE`, corresponding to  `os_log`'s `OS_LOG_TYPE_DEFAULT` level always hits disk. (Same for errors, which additionally also commit the info ones, and are specially flagged as important.)

As there's been some confusion over this a few times now, I think I might change the `KEXTLOG_NOTE` level to `KEXTLOG_DEFAULT` to match `os_log` and additionally put some comments in the enum explaining the memory vs disk syslog policy.